analysers/analyser_osmosis_highway_deadend.py

```diff
@@ -71,121 +71,227 @@
     bicycle_parking.id IS NULL
 """
 
+
 sql30 = """
-CREATE TEMP TABLE oneway AS
+-- Collect all oneways
+CREATE TEMP TABLE oneway_highway AS
 SELECT
   id,
-  t.nid,
+  nodes,
+  tags?'oneway' AND tags->'oneway' = '-1' AS oneway_reverse,
+  linestring,
+  array_length(nodes, 1) AS length,
+  generate_subscripts(nodes, 1) AS nid_index
+FROM
+  highways
+WHERE
+  (is_oneway OR is_roundabout) AND
+  NOT is_area AND
+  NOT is_construction
+"""
+
+sql31 = """
+CREATE INDEX idx_oneway_highway_linestring ON oneway_highway USING GIST(linestring);
+"""
+
+sql32 = """
+-- Collect all nodes that could be possible (valid) end nodes
+CREATE TEMP TABLE allowed_end_nodes AS
+SELECT DISTINCT
+  unnest(nodes) AS id
+FROM
+  ways
+WHERE
+  ways.tags != '' AND
+  (
+    (ways.tags?'amenity' AND ways.tags->'amenity' = 'parking') OR
+    (ways.tags?'railway' AND ways.tags->'railway' = 'platform') OR
+    (ways.tags?'aeroway' AND ways.tags->'aeroway' = 'taxiway')
+  )
+UNION ALL
+SELECT
+  id
+FROM
+  nodes
+WHERE
+  tags != '' AND
+  (
+    (tags?'amenity'  AND tags->'amenity' = 'parking_entrance') OR
+    (tags?'entrance' AND tags->'entrance'!= 'garage')
```

Comment:
```suggestion
    (tags?'entrance' AND tags->'entrance' != 'garage')
```

Reply:
Thanks! It's actually a typo, I wanted to retain the old behavior of the analyser on `entrance` nodes, but accidentally typed `!=` instead of `=`. Fixed.  (I also decided to whitelist entrance=emergency when fixing this: I don't want to encourage people to draw explicit paths to emergency exits while commonly there is no defined path to such an exit in the inside of a building, e.g. in a hall where they're "just along the sides". Drawing explicit paths that you don't see in the real location isn't good)